src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/TempDataSerializer.cs

```diff
@@ -0,0 +1,233 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.Internal;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Bson;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
+{
+    public class TempDataSerializer
+    {
+        private readonly JsonSerializer _jsonSerializer = JsonSerializer.Create(
+            new JsonSerializerSettings()
```

Comment:
@kichalla why make such a distinction? Use the secure settings everywhere. I left this use of the default settings alone in the past but can't justify the difference now.

@davidfowl we do pool `JsonSerializer` instances for use in `JsonInputFormatter`. But that should not be necessary as long as this class (like `SessionStateTempDataProvider` before it) only has one or two instances.